File: example/pingpong.py
import os
import sys
sys.path.insert(1, os.path.realpath('../python'))

import time

def ping(data = ''):
  print('Python: ping')
  time.sleep(1)
  fb.send('pong')

from filebus import FileBus
fb = FileBus('inout.txt', 'inout.txt')
fb.on('ping', ping)

# simply wait for JS handshake

File: example/pingpong.js
const FileBus = require('../');

const fb = new FileBus('inout.txt', 'inout.txt');

fb.on('pong', () => {
  console.log('JS: pong');
  setTimeout(() => fb.send('ping'), 1000);
});

fb.once('handshake', () => {
  // send the first ping
  fb.send('ping');
});

fb.handshake();
